lib/VMwareWebService/broker_timeout.rb

```diff
@@ -0,0 +1,64 @@
+module Timeout
```

Comment:
Won't the GVL prevent `kill` and `raise` to ever be executed simultaneously?

Also, couldn't thread x call `kill` on thread `y` after the yield, but before `state_lock` is synchronized?  Then we have the same situation?